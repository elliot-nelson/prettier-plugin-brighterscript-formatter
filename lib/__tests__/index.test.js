const prettier = require('prettier');

const uglyTerraform = `
module "some_module" {
  var1 = 3
  var2         = 4
  var3  =5
}
`;

describe('inferred parser', () => {
  const $env = process.env;

  beforeEach(() => {
    process.env = { ...$env };
  });

  afterEach(() => {
    process.env = $env;
  });

  it('automatically formats Terraform', () => {
    expect(
      prettier.format(uglyTerraform, {
        filepath: 'main.tf',
        plugins: ['.'],
      })
    ).toMatchSnapshot();
  });

  it('when no Terraform exists on path, fails to format and returns original text', () => {
    process.env.PATH = '';

    expect(
      prettier.format(uglyTerraform, {
        filepath: 'main.tf',
        plugins: ['.'],
      })
    ).toMatchSnapshot();
  });

  it('when no Terraform exists on path and terraformStrictError=true, fails', () => {
    process.env.PATH = '';

    expect(() =>
      prettier.format(uglyTerraform, {
        filepath: 'main.tf',
        plugins: ['.'],
        terraformStrictError: true,
      })
    ).toThrowError(/terraform: No such file or directory/);
  });
});
