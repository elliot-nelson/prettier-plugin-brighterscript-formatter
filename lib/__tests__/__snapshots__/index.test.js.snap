// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`inferred parser automatically formats Terraform 1`] = `
"
module "some_module" {
  var1 = 3
  var2 = 4
  var3 = 5
}
"
`;

exports[`inferred parser when no Terraform exists on path, fails to format and returns original text 1`] = `
"
module "some_module" {
  var1 = 3
  var2         = 4
  var3  =5
}
"
`;
